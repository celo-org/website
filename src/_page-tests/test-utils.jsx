import { render } from "@testing-library/react";
import * as React from "react";
import i18nForTests from "src/utils/i18nForTests";
import { I18nextProvider } from "react-i18next";
export function TestProvider({ children }) {
    return <I18nextProvider i18n={i18nForTests}>{children}</I18nextProvider>;
}
export const renderIgnoringUnstableFlushDiscreteUpdates = (component) => {
    const originalError = console.error;
    const error = jest.fn();
    console.error = error;
    const result = render(component);
    expect(error).toHaveBeenCalledTimes(1);
    expect(error).toHaveBeenCalledWith("Warning: unstable_flushDiscreteUpdates: Cannot flush updates when React is already rendering.%s", expect.any(String));
    console.error = originalError;
    return result;
};
//# sourceMappingURL=test-utils.jsx.map