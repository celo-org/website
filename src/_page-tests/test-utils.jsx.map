{"version":3,"file":"test-utils.jsx","sourceRoot":"","sources":["test-utils.tsx"],"names":[],"mappings":"AAAA,OAAO,EAAE,MAAM,EAAE,MAAM,wBAAwB,CAAA;AAC/C,OAAO,KAAK,KAAK,MAAM,OAAO,CAAA;AAC9B,OAAO,YAAY,MAAM,wBAAwB,CAAA;AAIjD,OAAO,EAAE,eAAe,EAAE,MAAM,eAAe,CAAA;AAI/C,MAAM,UAAU,YAAY,CAAC,EAAE,QAAQ,EAAS;IAC9C,OAAO,CAAC,eAAe,CAAC,IAAI,CAAC,CAAC,YAAY,CAAC,CAAC,CAAC,QAAQ,CAAC,EAAE,eAAe,CAAC,CAAA;AAC1E,CAAC;AAID,MAAM,CAAC,MAAM,0CAA0C,GAAG,CAAC,SAA6B,EAAE,EAAE;IAE1F,MAAM,aAAa,GAAG,OAAO,CAAC,KAAK,CAAA;IACnC,MAAM,KAAK,GAAG,IAAI,CAAC,EAAE,EAAE,CAAA;IACvB,OAAO,CAAC,KAAK,GAAG,KAAK,CAAA;IACrB,MAAM,MAAM,GAAG,MAAM,CAAC,SAAS,CAAC,CAAA;IAChC,MAAM,CAAC,KAAK,CAAC,CAAC,qBAAqB,CAAC,CAAC,CAAC,CAAA;IACtC,MAAM,CAAC,KAAK,CAAC,CAAC,oBAAoB,CAChC,iGAAiG,EACjG,MAAM,CAAC,GAAG,CAAC,MAAM,CAAC,CACnB,CAAA;IACD,OAAO,CAAC,KAAK,GAAG,aAAa,CAAA;IAE7B,OAAO,MAAM,CAAA;AACf,CAAC,CAAA"}