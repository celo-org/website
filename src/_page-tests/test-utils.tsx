import { render } from "@testing-library/react"
import * as React from "react"
import i18nForTests from "src/utils/i18nForTests"
interface Props {
  children?: React.ReactNode
}
import { I18nextProvider } from "react-i18next"



export function TestProvider({ children }: Props) {
  return <I18nextProvider i18n={i18nForTests}>{children}</I18nextProvider>
}

// https://github.com/testing-library/react-testing-library/issues/470
// an issue with muted on video causing a rerender, this seems to be be best solution for now
export const renderIgnoringUnstableFlushDiscreteUpdates = (component: React.ReactElement) => {
  // tslint:disable: no-console
  const originalError = console.error
  const error = jest.fn()
  console.error = error
  const result = render(component)
  expect(error).toHaveBeenCalledTimes(1)
  expect(error).toHaveBeenCalledWith(
    "Warning: unstable_flushDiscreteUpdates: Cannot flush updates when React is already rendering.%s",
    expect.any(String)
  )
  console.error = originalError
  // tslint:enable: no-console
  return result
}
