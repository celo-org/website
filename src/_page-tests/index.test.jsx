import HomePage from "pages/index";
import * as React from "react";
import * as renderer from "react-test-renderer";
import { TestProvider } from "src/_page-tests/test-utils";
import { BLOCKS } from "@contentful/rich-text-types";
function blurbFactory(unique) {
    return {
        sys: {
            id: unique,
            type: "",
            locale: "en",
            createdAt: "2021-07-01",
            updatedAt: "2021-07-01",
            contentType: { sys: { id: "proposition", type: "Link", linkType: "ContentType" } },
        },
        fields: {
            isNaturalSize: false,
            newIcon: false,
            title: `Blurb ${unique}`,
            body: {
                nodeType: BLOCKS.DOCUMENT,
                content: [
                    {
                        nodeType: BLOCKS.PARAGRAPH,
                        content: [{ nodeType: "text", value: "Hello World", marks: [], data: {} }],
                        data: {},
                    },
                ],
                data: {},
            },
        },
        toPlainObject: () => this,
        update: () => this,
    };
}
const TestData = {
    title: "Celo Home",
    description: "A description of Celo",
    slug: "home",
    darkNav: false,
    sections: [
        {
            sys: {
                id: "1",
                type: "",
                locale: "en",
                createdAt: "2021-07-01",
                updatedAt: "2021-07-01",
                contentType: { sys: { id: "grid-row", type: "Link", linkType: "ContentType" } },
            },
            fields: {
                id: "test",
                columns: 3,
                cells: [blurbFactory("romeo"), blurbFactory("soma"), blurbFactory("rico")],
            },
            toPlainObject: () => this,
            update: () => this,
        },
    ],
};
describe("HomePage", () => {
    it("renders", async () => {
        const tree = renderer
            .create(<TestProvider>
          <HomePage darkNav={false} title={TestData.title} description={TestData.description} slug={TestData.slug} sections={TestData.sections}/>
        </TestProvider>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
//# sourceMappingURL=index.test.jsx.map