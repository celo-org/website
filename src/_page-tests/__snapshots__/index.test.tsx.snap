// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage renders 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: hidden;
  max-width: 100vw;
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: hidden;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  width: 100%;
  background-color: #FBF6F1;
}

.emotion-2 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-left: 12px;
  padding-right: 12px;
  width: 100%;
  max-width: 100vw;
  overflow: hidden;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-column-gap: 24px;
  column-gap: 24px;
}

@media (min-width: 576px) and (max-width: 992px) {
  .emotion-2 {
    display: grid;
    grid-auto-rows: auto;
    -webkit-align-self: center;
    -ms-flex-item-align: center;
    align-self: center;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    padding-right: 24px;
    padding-left: 24px;
    width: 100%;
    max-width: 982px;
  }
}

@media (min-width: 992px) {
  .emotion-2 {
    display: grid;
    grid-auto-rows: auto;
    -webkit-align-self: center;
    -ms-flex-item-align: center;
    align-self: center;
    width: 100%;
    max-width: 1104px;
  }
}

@media (min-width: 992px) {
  .emotion-2 {
    grid-template-columns: 1fr 1fr 1fr;
  }
}

@media (min-width: 576px) and (max-width: 992px) {
  .emotion-2 {
    grid-template-columns: 1fr 1fr 1fr;
  }
}

.emotion-3 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  margin-bottom: 36px;
}

@media (max-width: 576px) {
  .emotion-3 {
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-align-content: center;
    -ms-flex-line-pack: center;
    align-content: center;
  }
}

.emotion-4 {
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.emotion-5 {
  font-weight: 200;
  margin: 0;
  margin-block-start: 0;
  margin-block-end: 0;
  font-family: GT-Alpina,serif;
  font-size: 24px;
  line-height: 28;
  text-rendering: geometricPrecision;
  color: #111214;
  line-height: 28px;
  margin-top: 16px;
  margin-bottom: 12px;
}

.emotion-6 {
  font-family: Inter,sans-serif;
  font-size: 20px;
  line-height: 28;
  text-rendering: geometricPrecision;
  color: #111214;
  line-height: 28px;
}

<div
  className="emotion-0"
>
  <section
    className="emotion-1"
  >
    <div
      className="emotion-2"
      id="test"
    >
      <div
        className="emotion-3"
      >
        <div
          className="emotion-4"
        >
          <h4
            className="emotion-5"
          >
            Blurb romeo
          </h4>
          <p
            className="emotion-6"
          >
            Hello World
          </p>
        </div>
      </div>
      <div
        className="emotion-3"
      >
        <div
          className="emotion-4"
        >
          <h4
            className="emotion-5"
          >
            Blurb soma
          </h4>
          <p
            className="emotion-6"
          >
            Hello World
          </p>
        </div>
      </div>
      <div
        className="emotion-3"
      >
        <div
          className="emotion-4"
        >
          <h4
            className="emotion-5"
          >
            Blurb rico
          </h4>
          <p
            className="emotion-6"
          >
            Hello World
          </p>
        </div>
      </div>
    </div>
  </section>
</div>
`;
